### PR TITLE
feat: authentication added login, register, logout

### PR DESCRIPTION
**Pull request by sonishreyas**
# Change logs

- Added Auth features
- login
- register
- logout
- Added navlinks to headers
- Added Profile page to logout

# Deploy Preview
https://deploy-preview-2--ripple-habits.netlify.app/

### CODE REVIEW

File: src/components/Authentication/Register.jsx [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+							false,
+							"UPDATE_FOCUS",
+							registerDispatch,
+							focusReset
+						)
+					}
+				/>
+				<label htmlFor="first-name" className="textbox-label m-0 px-1">
+					{" "}
+					First Name<span className="required-field">*</span>
```
Comment (Sharmaryan) [inline src/components/Authentication/Register.jsx:78]: remove {" "} these

File: src/utils/authentication.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
+import axios from "axios";
+import { Navigate, useLocation } from "react-router-dom";
+import { useAuth } from "../context";
+const RequireAuth = ({ children }) => {
+	const { authState } = useAuth();
+	const location = useLocation();
+	console.log(location);
+	return authState.token ? (
```
Comment (Sharmaryan) [inline src/utils/authentication.js:8]: remove console.log

**Review by Sharmaryan — APPROVED** [review]
UI is clean, Code is well structured, also folder structure is good

**Review by sach10-create — APPROVED** [review]
Remove commented and console.logs.
Good to merger

**Pull request merged**